### PR TITLE
layers: Fix mem leak

### PR DESCRIPTION
**Pull request by juan-lunarg**

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -4809,6 +4809,7 @@ void CoreChecks::PreCallRecordDestroyDevice(VkDevice device, const VkAllocationC
 
         if (result != VK_SUCCESS) {
             LogInfo(device, "UNASSIGNED-cache-retrieval-error", "Validation Cache Retrieval Error");
+            free(validation_cache_data);
```
Comment (juan-lunarg) [inline layers/core_validation.cpp:4812]: clang-tidy found this minor issue. Where we return early on an error and don't free the memory.

**Review by Tony-LunarG — APPROVED** [review]
LGTM

**Review by jeremyg-lunarg — APPROVED** [review]
Fix is correct but could we use std::vector<uint8_t> or some other RAII object to avoid the need for freeing at all?

**Comment by juan-lunarg** [comment]
> Fix is correct but could we use std::vector<uint8_t> or some other RAII object to avoid the need for freeing at all?

My plan is to address all the malloc/free calls in VVL in another PR.

**Pull request merged**